gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -190,6 +215,16 @@ private void submit(Job job) throws ClassNotFoundException, IOException, Interru
     }
   }
 
+  private void markOutputDirAsCompleted(DateTime jobStartTime) throws IOException {
+    Path completionFilePath = new Path(this.outputPath, ConfigurationKeys.COMPACTION_COMPLETE_FILE_NAME);
+    FSDataOutputStream completionFileStream = this.fs.create(completionFilePath);
```

Comment:
`this.fs.create` should be in the `try` block, otherwise if it fails, it is possible that `completionFileStream` is an open stream that is never closed.

Please use `Closer` which is consistently used in Gobblin to close resources.